flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TablePipeline.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * This interface describes a pipeline created from a single {@code INSERT INTO} statement, created
+ * either using Table APIs or SQL.
+ */
+@PublicEvolving
+public interface TablePipeline extends Explainable<TablePipeline>, Executable, Compilable {}
```

Comment:
to fill this interface with something useful. how about we provide an `ObjectIdentifier` for the sink table.

Reply:
I added both `getSinkIdentifier` and implemented `toString` in `TablePipelineImpl`